pkg/controller/storagecluster/reconcile.go

```diff
@@ -727,12 +732,30 @@ func newCephCluster(sc *ocsv1.StorageCluster, cephImage string, nodeCount int) *
 			ContinueUpgradeAfterChecksEvenIfNotHealthy: true,
 		},
 	}
-
-	if len(sc.Spec.MonDataDirHostPath) != 0 {
-		cephCluster.Spec.DataDirHostPath = sc.Spec.MonDataDirHostPath
-	}
-	if sc.Spec.MonPVCTemplate != nil {
-		cephCluster.Spec.Mon.VolumeClaimTemplate = sc.Spec.MonPVCTemplate
+	monPVCTemplate := sc.Spec.MonPVCTemplate
+	monDataDirHostPath := sc.Spec.MonDataDirHostPath
+	// If the `monPVCTemplate` is provided, the mons will provisioned on the
+	// provided `monPVCTemplate`.
+	if monPVCTemplate != nil {
+		cephCluster.Spec.Mon.VolumeClaimTemplate = monPVCTemplate
+		// If the `monDataDirHostPath` is provided without the `monPVCTemplate`,
+		// the mons will be provisioned on the provided `monDataDirHostPath`.
+	} else if len(monDataDirHostPath) > 0 {
+		cephCluster.Spec.DataDirHostPath = monDataDirHostPath
+		// If no `monPVCTemplate` and `monDataDirHostPath` is provided, the mons will
+		// be provisioned using the PVC template of first StorageDeviceSets if present.
+	} else if len(sc.Spec.StorageDeviceSets) > 0 {
+		ds := sc.Spec.StorageDeviceSets[0]
+		cephCluster.Spec.Mon.VolumeClaimTemplate = &corev1.PersistentVolumeClaim{
+			Spec: corev1.PersistentVolumeClaimSpec{
+				StorageClassName: ds.DataPVCTemplate.Spec.StorageClassName,
+				Resources: corev1.ResourceRequirements{
+					Requests: corev1.ResourceList{
+						corev1.ResourceStorage: resource.MustParse("10Gi"),
+					},
+				},
+			},
+		}
```

Comment:
You did not add the warning that was requested.

Reply:
its there https://github.com/openshift/ocs-operator/pull/410/files#diff-24bbbd5f2f20ffccf914757966c7401aR590-R592